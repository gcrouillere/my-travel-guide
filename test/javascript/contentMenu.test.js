import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router'

import ContentMenu from '../../app/javascript/packs/articleForm/contentMenu'
import MapInitialCenterOverlay from '../../app/javascript/packs/articleForm/contentMenu/mapInitialCenterOverlay'
import PhotoInitialFileOverlay from '../../app/javascript/packs/articleForm/contentMenu/photoInitialFileOverlay'

configure({ adapter: new Adapter() });

describe('ContentMenu test suite', () => {
  let addNewTextContent, addNewMap, addNewPhotoBloc, addNewComponentOnDrag = jest.fn()
  addNewTextContent = addNewMap = addNewPhotoBloc = addNewComponentOnDrag = jest.fn()

  const wrapper = mount(<ContentMenu addNewTextContent={addNewTextContent}
    addNewPhotoBloc={addNewPhotoBloc} addNewMap={addNewMap} id={1} addNewComponentOnDrag={addNewComponentOnDrag}/>)

  it('displays Content Menu', () => {
    expect(wrapper.find('.buttons .blocAddition').length).toEqual(3)
    expect(wrapper.find('.expand').length).toEqual(1)
  })

  it('displays photo overlay on click', () => {
    wrapper.find('.buttons .photo .addNew').simulate('click')
    expect(wrapper.find(PhotoInitialFileOverlay).prop('photoOverlayActive')).toEqual(true)
  })

  it('passes photo addition to upper component', () => {
    wrapper.instance().addNewPhotoBloc()
    expect(addNewPhotoBloc).toHaveBeenCalled();
  })

  it('displays map overlay on click', () => {
    wrapper.find('.buttons .map .addNew').simulate('click')
    expect(wrapper.find(MapInitialCenterOverlay).prop('mapOverlayActive')).toEqual(true)
  })

  it('defines map characteristics before creation', () => {
    wrapper.instance().addNewMap("mapLocation", 2)
    expect(addNewMap).toHaveBeenCalledWith(1, {lat: 0, lng: 0}, "mapLocation", 2)
  })

  it('passes text bloc addition to upper component', () => {
    wrapper.find('.buttons .text .addNew').simulate('click')
    expect(addNewTextContent).toHaveBeenCalled();
    expect(wrapper.state.initPositionAtCreation).toEqual(undefined)
  })

  it('triggers proper component creation on drag', () => {
    wrapper.find(".buttons .map .addNew").simulate('dragstart')
    expect(addNewComponentOnDrag).toHaveBeenCalled();
    expect(addNewComponentOnDrag).toHaveBeenCalledWith(expect.anything(), "mapCreation");
  })

  it('sets overlay to inactive on function call', () => {
    wrapper.instance().abandonMapCreation()
    expect(wrapper.state('mapOverlayActive')).toEqual(false)
    wrapper.instance().abandonPhotoCreation()
    expect(wrapper.state('photoOverlayActive')).toEqual(false)
  })

  it('gets location from lower component', () => {
    let event = {target: {value: "foo"}}
    wrapper.instance().printLocation(event)
    expect(wrapper.state('location')).toEqual("foo")
  })
})

