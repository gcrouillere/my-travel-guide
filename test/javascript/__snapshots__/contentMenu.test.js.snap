// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="contentMenu"
  id="contentMenu"
>
  <div
    className="buttons"
  >
    <div
      className="blocAddition text"
    >
      <div
        className="btn btn-dark addNew"
        draggable={true}
        onClick={[Function]}
        onDragStart={[Function]}
        value="text"
      >
        <div
          className="addIcone"
          draggable={false}
        >
          <img
            src="test-file-stub"
          />
        </div>
        <div
          className="addExplain"
        >
          <div>
            Click to add text in queue
          </div>
          <div>
            Drag to add text to a specific location
          </div>
        </div>
      </div>
    </div>
    <div
      className="blocAddition map"
    >
      <div
        className="btn btn-dark addNew"
        draggable={true}
        onClick={[Function]}
        onDragStart={[Function]}
        value="map"
      >
        <div
          className="addIcone"
          draggable={false}
        >
          <img
            src="test-file-stub"
          />
        </div>
        <div
          className="addExplain"
        >
          <div>
            Click to add map in queue
          </div>
          <div>
            Drag to add map to a specific location
          </div>
        </div>
      </div>
    </div>
    <div
      className="blocAddition photo"
    >
      <div
        className="btn btn-dark addNew"
        draggable={true}
        onClick={[Function]}
        onDragStart={[Function]}
        value="photo"
      >
        <div
          className="addIcone"
          draggable={false}
        >
          <img
            src="test-file-stub"
          />
        </div>
        <div
          className="addExplain"
        >
          <div>
            Click to add photo in queue
          </div>
          <div>
            Drag to add photo to a specific location
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="expand"
  >
    <div
      className="menu"
    >
      <p>
        M
      </p>
      <p>
        E
      </p>
      <p>
        N
      </p>
      <p>
        U
      </p>
    </div>
    <div
      className="chevron-closed"
    >
      &lt;
    </div>
    <div
      className="chevron-open"
    >
      &gt;
    </div>
  </div>
  <div
    className="mapInitialCenterOverlay "
  >
    <button
      aria-label="Close"
      className="close"
      onClick={[Function]}
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
    <p
      className="mapOverlayTitle"
    >
      Which area do you want to show ?
    </p>
    <p
      className="mapOverlaydescription"
    >
      If no suggestion is given, try a near match. You 'll be able to fine tune the map in the article editor.
    </p>
    <div
      className="input-group input-group-lg"
    >
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
          id="inputGroup-sizing-large"
        >
          Map's center
        </span>
      </div>
      <input
        aria-describedby="inputGroup-sizing-large"
        aria-label="Large"
        className="form-control"
        id="initialMapLocation"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="photoInitialFileOverlay "
  >
    <button
      aria-label="Close"
      className="close"
      onClick={[Function]}
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
    <p
      className="mapOverlayTitle"
    >
      Select a photo on your disk
    </p>
    <div
      className="input-group mb-3"
    >
      <div
        className="custom-file"
      >
        <input
          accept="image/*"
          aria-describedby="inputGroupFileAddon01"
          className="custom-file-input"
          id="fileupload"
          multiple={false}
          onChange={[Function]}
          type="file"
        />
        <label
          className="custom-file-label"
          htmlFor="fileupload"
        >
          Click here to browse your files
        </label>
      </div>
    </div>
  </div>
</div>
`;
